tests/plot_sync/test_receiver.py

```diff
@@ -247,10 +251,24 @@ async def test_to_dict(counts_only: bool) -> None:
     # But unequal dicts wit the opposite counts_only value
     assert plot_sync_dict_1 != receiver.to_dict(not counts_only)
 
-    # Walk through all states from idle to done and run them with the test data
+    expected_plot_files_processed: int = 0
+    expected_plot_files_total: int = sync_steps[State.idle].args[2]
+
+    # Walk through all states from idle to done and run them with the test data and validate the sync progress
     for state in State:
         await run_sync_step(receiver, sync_steps[state], state)
 
+        if state != State.idle and state != State.removed and state != State.done:
+            expected_plot_files_processed += len(sync_steps[state].args[0])
+
+        sync_data = receiver.to_dict()["syncing"]
+        if state == State.done:
+            assert sync_data is None
+        else:
+            assert sync_data["initial"]
+            assert sync_data["plot_files_processed"] == expected_plot_files_processed
+            assert sync_data["plot_files_total"] == expected_plot_files_total
```

Comment:
One assertion with all the information when there's an error, verifies that `"initial"` is `True` not `[1, 2, 3]` or `"abc"`, checks there aren't extra keys.

```suggestion
        if state == State.done:
            expected_sync_data = None
        else:
            expected_sync_data = {
                "initial": True,
                "plot_files_processed": expected_plot_files_processed,
                "plot_files_total": expected_plot_files_total,
            }

        assert sync_data == expected_sync_data
```